### ISSUE TITLE
Is FLUX.1-Krea-dev supported to train a LoRa?

### ISSUE THREAD

**Issue by kabalao** [ostris/ai-toolkit#364]
Thanks.

**Comment by Poukpalaova** [comment]
I'm about to test is using a flux dev pipeline, we will see. I only change the model path to the new krea one in the config file: name_or_path: "black-forest-labs/FLUX.1-Krea-dev". I'll tell if it doesn't work.

**Comment by StefanRademakers** [comment]
@Poukpalaova Very curious about this too! Was going to test the same thing.

**Comment by klausxp** [comment]
Has it finally worked? Best regards!

**Comment by berradakamal** [comment]
I tested four different models, running them from 3,000 to 4,000 steps. While some images turned out well, only about one out of four models produced good results — the rest came out blurry and pixelated, even though the dataset quality wasn’t bad.
```---
job: extension
config:
  # this name will be the folder and filename name
  name: "jason"
  process:
    - type: 'sd_trainer'
      # root folder to save training sessions/samples/weights
      training_folder: "output"
      # uncomment to see performance stats in the terminal every N steps
#      performance_log_every: 1000
      device: cuda:0
      # if a trigger word is specified, it will be added to captions of training data if it does not already exist
      # alternatively, in your captions you can add [trigger] and it will be replaced with the trigger word
      trigger_word: "jason"
      network:  
        type: "lora"
        linear: 16
        linear_alpha: 16
      save:
        dtype: float16 # precision to save
        save_every: 200 # save every this many steps
        max_step_saves_to_keep: 4 # how many intermittent saves to keep
        push_to_hub: false # change this to True to push your trained model to Hugging Face.
        # You can either set up a HF_TOKEN env variable or you'll be prompted to log-in         
#       hf_repo_id: your-username/your-model-slug
#       hf_private: true # whether the repo is private or public
      datasets:
       # datasets are a folder of images. Captions need to be txt files with the same name as the image
        # for instance image2.jpg and image2.txt. Only jpg, jpeg, and png are supported currently
        # images will automatically be resized and bucketed into the resolution specified
        - folder_path: "/dataset"
          caption_ext: "txt"
          caption_dropout_rate: 0.05  # will drop out the caption 5% of time
          shuffle_tokens: false  # shuffle caption order, split by commas
          cache_latents_to_disk: true  # leave this true unless you know what you're doing
          resolution: [ 512 ]  # flux enjoys multiple resolutions
      train:
        batch_size: 1
        steps: 3000  # total number of steps to train 500 - 4000 is a good range
        gradient_accumulation_steps: 1
        train_unet: true
        train_text_encoder: false  # probably won't work with flux
        gradient_checkpointing: true  # need this on unless you have a ton of VRAM
        noise_scheduler: "flowmatch" # for training only
        optimizer: "adamw8bit"
        lr: 1e-4
        # uncomment this to skip the pre-training sample
#        skip_first_sample: true
        # uncomment to completely disable sampling
#        disable_sampling: true
        # uncomment to use new vell curved weighting. Experimental but may produce better results
#        linear_timesteps: true

        # ema will smooth out learning, but could slow it down. Recommended to leave on.
  ema_config:
          use_ema: true
          ema_decay: 0.99

        # will probably need this if gpu supports it for flux, other dtypes may not work correctly
        dtype: bf16
      model:
        # huggingface model name or path
        name_or_path: "black-forest-labs/FLUX.1-Krea-dev"
        is_flux: true
        quantize: true  # run 8bit mixed precision
        low_vram: true  # uncomment this if the GPU is connected to your monitors. It will use less vram to quantize, but is slower.
      sample:
        sampler: "flowmatch" # must match train.noise_scheduler
        sample_every: 500 # sample every this many steps
        width: 512
        height: 512
        prompts:
          - "portrait of [jason], close-up, day light, high resolution"
          - "Generate a minimalist face portrait, hand-drawn line art illustration of [jason]. Use bold, clean lines with>
        neg: ""  # not used on flux
        seed: 42
        walk_seed: true
        guidance_scale: 3.5
        sample_steps: 20
# you can add any additional meta info here. [name] is replaced with config name at top
meta:
  name: "jason"
 version: '1.0'

```

**Comment by Wallawalla47** [comment]
> I tested four different models, running them from 3,000 to 4,000 steps. While some images turned out well, only about one out of four models produced good results — the rest came out blurry and pixelated, even though the dataset quality wasn’t bad.

I had the same issues. Not sure what's going on.  I was reading somewhere that apparently Flux Krea likes a lower training rate and a lower alpha than the original Flux Dev, but not sure if that's the cause of this problem.

**Comment by Poukpalaova** [comment]
I tryed 2 loras, 1 detailer and a person. I noticed that the normal 1e-4 is too fast for krea lr. 2e-5, 3e-5 is better, at 4000 steps. I was at 32 alpha so i'll try the suggestion of @berradakamal  to lower it.

**Comment by Poukpalaova** [comment]
> [@Poukpalaova](https://github.com/Poukpalaova) Very curious about this too! Was going to test the same thing.

that worked

**Comment by berradakamal** [comment]
I'll try to lower it and train some new loras today with different configs as suggested. Will keep you posted

**Comment by berradakamal** [comment]
@Poukpalaova with a lower lr your loras were better?

**Comment by svnaxis** [comment]
Tried with 3e-5 LR. Loras trained on vanilla FLUX still looks infinitely better than the ones trained on Krea...

**Comment by Poukpalaova** [comment]
> [@Poukpalaova](https://github.com/Poukpalaova) with a lower lr your loras were better?

Yes. I even tryed to start a training with Krea for 2000 steps, then i reused same yaml but changed the model to flux dev and step to 4000 to continue with flux dev, but should have lowered the lr way lower (to do), But it turned good too. at around 2200 steps it was even good.

**Comment by klausxp** [comment]
Hello, good morning!

I’ve been training a LoRA on Flux Krea using Kohya (without training the text encoder), and I’ve noticed that the Trigger Word barely responds, and the likeness/fidelity is not what I expected.

During training, some of the automatically generated preview images looked interesting, but when testing the final LoRAs, the model didn’t really follow the trigger. I’ve tested up to step 3000 with bf16, 3e-5 learning rate, 64 rank, and 32 alpha.

I’ve seen people getting very good results, but I haven’t found anyone sharing concrete details like:

Full training configuration (LR, alpha, steps, batch size, etc.).

Whether they trained the text encoder or not.

Exact caption format (tags vs natural sentences, word count, whether the trigger is included in every image).

Dataset size and quality (number of images, styles included, resolution, horizontal vs vertical proportions).

Which tool or pipeline was used (Kohya, ComfyUI, AI-Toolkit, SwarmUI, etc.).

If anyone could share their best tested configuration for training a LoRA for Flux Krea and a trigger word, it would be great to compare and keep testing.

Thanks in advance 🙏

**Comment by kremenevskiy** [comment]
Very waiting for krea support and good configuration for it to train persons images. Thank you :)

**Comment by berradakamal** [comment]
Tried today with following: 
```---
job: extension
config:
  # this name will be the folder and filename name
  name: "test-4"
  process:
    - type: 'sd_trainer'
      # root folder to save training sessions/samples/weights
      training_folder: "output"
      # uncomment to see performance stats in the terminal every N steps
#      performance_log_every: 1000
      device: cuda:0
      # if a trigger word is specified, it will be added to captions of training data if it does not alrea>
      # alternatively, in your captions you can add [trigger] and it will be replaced with the trigger word
      trigger_word: "test-4"
      network:
        type: "lora"
        linear: 16
        linear_alpha: 16
      save:
        dtype: float16 # precision to save
        save_every: 200 # save every this many steps
        max_step_saves_to_keep: 4 # how many intermittent saves to keep
        push_to_hub: false # change this to True to push your trained model to Hugging Face.
 datasets:
        # datasets are a folder of images. Captions need to be txt files with the same name as the image
        # for instance image2.jpg and image2.txt. Only jpg, jpeg, and png are supported currently
        # images will automatically be resized and bucketed into the resolution specified
        - folder_path: "/dataset"
          caption_ext: "txt"
          caption_dropout_rate: 0.05  # will drop out the caption 5% of time
          shuffle_tokens: false  # shuffle caption order, split by commas
          cache_latents_to_disk: true  # leave this true unless you know what you're doing
          resolution: [ 512 ]  # flux enjoys multiple resolutions
      train:
        batch_size: 1
        steps: 4000  # total number of steps to train 500 - 4000 is a good range
        gradient_accumulation_steps: 1
        train_unet: true
        train_text_encoder: false  # probably won't work with flux
        gradient_checkpointing: true  # need this on unless you have a ton of VRAM
        noise_scheduler: "flowmatch" # for training only
optimizer: "adamw8bit"
        lr: 2e-5
        # uncomment this to skip the pre-training sample
#        skip_first_sample: true
        # uncomment to completely disable sampling
#        disable_sampling: true
        # uncomment to use new vell curved weighting. Experimental but may produce better results
#        linear_timesteps: true

        # ema will smooth out learning, but could slow it down. Recommended to leave on.
        ema_config:
          use_ema: true
          ema_decay: 0.99

        # will probably need this if gpu supports it for flux, other dtypes may not work correctly
        dtype: bf16
      model:
        # huggingface model name or path
        name_or_path: "black-forest-labs/FLUX.1-Krea-dev"
        is_flux: true
  quantize: true  # run 8bit mixed precision
        low_vram: true  # uncomment this if the GPU is connected to your monitors. It will use less vram t>
      sample:
        sampler: "flowmatch" # must match train.noise_scheduler
        sample_every: 500 # sample every this many steps
        width: 512
        height: 512
        prompts:
          - "portrait of [test-4], close-up, day light, high resolution"
          - "Generate a minimalist face portrait, hand-drawn line art illustration of [test-4]. U>
        neg: ""  # not used on flux
        seed: 42
        walk_seed: true
        guidance_scale: 3.5
        sample_steps: 20
# you can add any additional meta info here. [name] is replaced with config name at top
meta:
  name: "test-4"
  version: '1.0'```

Results are still not accurate with the dataset, quality is not the same as the one I see on Reddit
